CRM/Utils/Request.php

```diff
@@ -88,10 +88,9 @@ public static function retrieve($name, $type, &$store = NULL, $abort = FALSE, $d
         break;
     }
 
-    if (isset($value) &&
-      (CRM_Utils_Type::validate($value, $type, $abort, $name) === NULL)
-    ) {
-      $value = NULL;
+    if (isset($value)) {
+      // The validate function will cast to numeric, if appropriate or set to NULL if the data is not valid.
+      $value = CRM_Utils_Type::validate($value, $type, $abort, $name);
```

Comment:
@eileenmcnaughton Can we pull this out and merge (first) separately? This *should* be safe but it is hit by a *lot* of code.

Reply:
ok

Reply:
https://github.com/civicrm/civicrm-core/pull/16525